### ISSUE TITLE
Packer's filetype is not unique enough and conflicts with terraform's packer filetype

### ISSUE THREAD

**Issue by akinsho** [wbthomason/packer.nvim#457]
So it seems that potentially the `packer` filetype might not be unique enough. There is further context in the issue I've linked, but there *might* be a programming language (terraform, ~not really a language 🤔~) with a filetype of `packer`. This was added to `nvim-treesitter` recently and is causing packers buffers to be treated as terraform which is drawing error highlights all over it, since it is not valid terraform.

https://github.com/nvim-treesitter/nvim-treesitter/issues/1524

I think it might be worth making packer's filetype more unique in either case, not sure if that will be a breaking change if any users depend on the packer filetype though. I should clarify that I believe this might be a bug on nvim-treesitter's side, so maybe we don't need to do anything.

cc @wbthomason

**Comment by wbthomason** [comment]
Thanks for catching this; I don't use `maintained` with `nvim-treesitter`, so I hadn't noticed any issues.

I would hazard a guess that the number of users depending on specifically the `packer` filetype is lower than the number of users installing all maintained TS parsers, so I'd favor a breaking change on our end if we have to choose.

Ideally, there would just be a way to disable TS in the `packer` buffer, rather than messing with the filetype either way. I don't know of any way to do this, but I'm not knowledgeable about TS.

**Comment by wbthomason** [comment]
Sorry, prematurely closed this. Thanks for catching it; it appears from the linked thread as though it will be (for now) resolved on `treesitter`'s side.

Still, might be good to move to a more unique name to prevent future trouble. Setting the filetype to `packer.nvim` should do it, I'd think?

**Comment by akinsho** [comment]
@wbthomason adding something nvim specific sounds like a good shout, not sure if `.` is allowed in a filetype though or if vim will behave strangely because of that? couldn't find any clarity in the vim docs (only looked briefly tbh)

**Comment by nanotee** [comment]
From `:help 'filetype'`:
> 	When a dot appears in the value then this separates two filetype
>	names.  Example:
>		/* vim: set filetype=c.doxygen : */ ~
>	This will use the "c" filetype first, then the "doxygen" filetype.

So I think `packer.nvim` would still load the runtime files associated with the `packer` filetype

**Comment by wbthomason** [comment]
@nanotee: Ah, damn. I guess that's why `pandoc` uses `markdown.pandoc`.

Perhaps `packer-nvim`? The dash doesn't appear in the illegal filetype name characters given in `:help 'filetype'`.

**Comment by akinsho** [comment]
I think hyphens are safe, well I've used underscore in another plugin and that is yet to explode, so I'm guessing/hoping that means hyphen is also safe.

**Comment by wbthomason** [comment]
Yes - I also tested it out and it works fine (and there's nothing in the docs indicating that a hyphenated filetype gets classified as two filetypes).
